lib/src/supabase_auth.dart

```diff
@@ -300,12 +300,101 @@ extension GoTrueClientSignInProvider on GoTrueClient {
       scopes: scopes,
       queryParams: queryParams,
     );
-    final url = Uri.parse(res.url!);
-    final result = await launchUrl(
-      url,
-      mode: authScreenLaunchMode,
-      webOnlyWindowName: '_self',
+    final uri = Uri.parse(res.url!);
+
+    bool? result;
+    if (authScreenLaunchMode == LaunchMode.inAppWebView) {
+      result = await showModalBottomSheet<bool>(
+        context: context,
+        isScrollControlled: true,
+        builder: (context) => _OAuthSignInWebView(
+          uri: uri,
+          redirectTo: redirectTo,
+        ),
+      );
+    } else {
+      result = await launchUrl(
+        uri,
+        mode: authScreenLaunchMode,
+        webOnlyWindowName: '_self',
+      );
+    }
+    return result ?? false;
+  }
+}
+
+class _OAuthSignInWebView extends StatefulWidget {
+  const _OAuthSignInWebView({
+    Key? key,
+    required this.uri,
+    required this.redirectTo,
+  }) : super(key: key);
+
+  final Uri uri;
+  final String? redirectTo;
+
+  @override
+  State<_OAuthSignInWebView> createState() => _OAuthSignInWebViewState();
+}
+
+/// Modal bottom sheet with webview for OAuth sign in
+class _OAuthSignInWebViewState extends State<_OAuthSignInWebView> {
+  late final WebViewController controller;
```

Comment:
It seems like this isn't being used, but is it?

Reply:
Yeah I forgot it when i downgraded the library. I’ll delete it